File: implementations/go/state/state_test.go
package state

import "testing"

func TestStateEventTypesIncludesAllRegistryEntries(t *testing.T) {
	expected := []string{
		"state.snapshot.requested",
		"state.snapshot.ready",
		"state.delta.applied",
		"state.invalidated",
		"freshness.expired",
		"freshness.window.changed",
	}
	for _, typ := range expected {
		if !EventTypes[typ] {
			t.Errorf("missing: %s", typ)
		}
	}
	if len(EventTypes) != len(expected) {
		t.Errorf("expected %d types, got %d", len(expected), len(EventTypes))
	}
}

func TestIsEventTypePositives(t *testing.T) {
	for _, typ := range []string{
		"state.snapshot.requested",
		"state.snapshot.ready",
		"state.delta.applied",
		"state.invalidated",
		"freshness.expired",
		"freshness.window.changed",
	} {
		if !IsEventType(typ) {
			t.Errorf("expected true for %s", typ)
		}
	}
}

func TestIsEventTypeNegatives(t *testing.T) {
	for _, typ := range []string{
		"task.submitted",
		"session.opened",
		"context.updated",
		"",
	} {
		if IsEventType(typ) {
			t.Errorf("expected false for %s", typ)
		}
	}
}

File: implementations/typescript/packages/state/test/state.test.js
import assert from "node:assert/strict";
import test from "node:test";
import { STATE_EVENT_TYPES, isStateEventType } from "../src/index.js";

test("state event types includes all registry entries", () => {
  const expected = [
    "state.snapshot.requested",
    "state.snapshot.ready",
    "state.delta.applied",
    "state.invalidated",
    "freshness.expired",
    "freshness.window.changed"
  ];
  for (const type of expected) {
    assert.equal(STATE_EVENT_TYPES.has(type), true, `missing: ${type}`);
  }
  assert.equal(STATE_EVENT_TYPES.size, expected.length);
});

test("isStateEventType positives", () => {
  assert.equal(isStateEventType("state.snapshot.requested"), true);
  assert.equal(isStateEventType("state.snapshot.ready"), true);
  assert.equal(isStateEventType("state.delta.applied"), true);
  assert.equal(isStateEventType("state.invalidated"), true);
  assert.equal(isStateEventType("freshness.expired"), true);
  assert.equal(isStateEventType("freshness.window.changed"), true);
});

test("isStateEventType negatives", () => {
  assert.equal(isStateEventType("task.submitted"), false);
  assert.equal(isStateEventType("session.opened"), false);
  assert.equal(isStateEventType("context.updated"), false);
  assert.equal(isStateEventType(""), false);
});
